src/pandablocks/hdf.py

```diff
@@ -100,8 +105,12 @@ def create_dataset(self, field: FieldCapture, raw: bool):
         else:
             # No processor, datatype passed through
             dtype = field.type
+        dataset_name = self.capture_record_hdf_names.get(field.name.replace(".", ":"), {}).get(
```

Comment:
Can we put the replace on the field name in the ioc level instead please? We don't do anything in the client with `:` otherwise, so seems neater there